Tests/Unit/EmogrifierTest.php

```diff
@@ -1090,4 +1090,45 @@ public function emogrifyMergesCssWithMixedUnits()
             $this->subject->emogrify()
         );
     }
+
+    /**
+     * @test
+     */
+    public function emogrifyRemovesDisplayNoneElements()
+    {
+        $css = 'div.foo { display: none; }';
```

Comment:
We also should have tests for both cases: When the display: none comes from a style node, and when it comes from a style attribute of the element.